code/game/objects/effects/effect_system/smoke.dm

```diff
@@ -174,6 +174,8 @@
 
 
 /obj/effect/particle_effect/smoke/tactical/proc/uncloak_smoke_act(var/mob/living/M)
+	if(!istype(M))
```

Comment:
This check should be happening in `/obj/effect/particle_effect/smoke/tactical/Uncrossed(var/mob/living/M)`
Which should actually be using `atom/movable/O` as per the reference, instead of assuming only `living` trigger it.
Same as `/obj/effect/particle_effect/smoke/tactical/Crossed(atom/movable/M)` really.